src/lib/parser/parser.ts

```diff
@@ -32,70 +32,42 @@ export class AuctionParser {
 		return processedMsg;
 	}
 
-	private wordIsAuctionType(word: string) {
-		const wtsAuctionTypes = ['WTS', 'SELLING'];
-		const wtbAuctionTypes = ['WTB', 'BUYING'];
-
-		// Check if the word includes any of the WTS auction types
-		for (const auctionType of wtsAuctionTypes) {
-			if (word.includes(auctionType)) {
-				return AuctionTypes.WTS;
-			}
-		}
-
-		// Check if the word includes any of the WTB auction types
-		for (const auctionType of wtbAuctionTypes) {
-			if (word.includes(auctionType)) {
-				return AuctionTypes.WTB;
+	// Thanks rici from StackOverflow for saving me time!
+	// Based on https://stackoverflow.com/a/30472781
+	private composeRanges(ranges: { start: number; end: number }[]) {

```

Comment:
I do kinda wish I could rewrite this function to be less "magical", it's the one part about this that is incomprehensible unless you know what it's doing already. I'm sure it could be written in a more sane style.